SteamBot/UserHandler.cs

```diff
@@ -190,6 +203,26 @@ public virtual void OnTradeRequestReply(bool accepted, string response)
 
         }
 
+        /// <summary>
+        /// Waits for the user to enter something into regular or trade chat, then returns it (as the result of a task)
+        /// Usage: The following displays "How many do you want to buy" and stores the user's response:
+        /// string userResponse = await GetUserResponse("How many do you want to buy?");
+        /// 
+        /// Note: calling this method causes the next user-message to NOT call OnMessage() or OnTradeMessage()
+        /// </summary>
+        /// <param name="message">An option message to display to the user.
+        /// Sent to whichever chat (normal or trade) is currently being used.</param>
+        protected virtual Task<string> GetUserResponse(string message = null)
+        {
+            if (message != null)
+            {
+                SendReplyMessage(message);
+            }
+
+            _waitingOnUserResponse = new TaskCompletionSource<string>();
```

Comment:
What are you talking about?  [`TaskCompletionSource`](https://msdn.microsoft.com/en-us/library/dd449174%28v=vs.110%29.aspx) is defined in `System.Threading.Tasks`, which is included.

https://github.com/Jessecar96/SteamBot/pull/705/files#diff-e23f3aec5a9b40a5175146cd6c380be5L5